LaraApi/src-lara/lara/vitishls/VitisHls.js

```diff
@@ -0,0 +1,153 @@
+        Clava.writeCode(weavingFolder);
+
+        // if no files were added, we assume that every woven file should be used
+        if (this.sourceFiles.length == 0) {
+            println("[" + this.toolName + "] No source files specified, assuming current AST is the input");
+            for (var file of Io.getFiles(Clava.getWeavingFolder())) {
+                const exts = [".c", ".cpp", ".h", ".hpp"];
+                const res = exts.some(ext => file.name.includes(ext));
+                if (res)
+                    str += "add_files " + weavingFolder + "/" + file.name + "\n"
+            }
+        }
+        else {
+            for (const file of this.sourceFiles) {
+                str += "add_files " + weavingFolder + "/" + file + "\n";
+            }
+        }
+        return str;
+    }
+
+    #generateTclFile() {
+        var cmd = "open_project " + this.vitisProjName + "\n";
+        cmd += "set_top " + this.topFunction + "\n";
+        cmd += this.#getTclInputFiles();
+        cmd += "open_solution \"solution1\" -flow_target vitis\n";
+        cmd += "set_part {" + this.platform + "}\n";
+        cmd += "create_clock -period " + this.clock + " -name default\n";
+        cmd += "csynth_design\n";
+        cmd += "exit\n";
+
+        Io.writeFile(this.vitisDir + "/script.tcl", cmd);
+    }
+
+    getSynthesisReport() {
+        println("[" + this.toolName + "] Processing synthesis report");
+
+        const parser = new VitisHlsReportParser(this.#getSynthesisReportPath());
+        const json = parser.getSanitizedJSON();
+
+        println("[" + this.toolName + "] Finished processing synthesis report");
+        return json;
+    }
+
+    preciseStr(n, decimalPlaces) {
+        return (+n).toFixed(decimalPlaces);
+    }
+
+    prettyPrintReport(report) {
+        println("----------------------------------------");
+        println("Vitis HLS synthesis report");
+        println("");
+        println("Targeted a " + report["platform"] + ", with target clock " + this.preciseStr(report["clockTarget"], 2) + "ns");
+        println("");
+        println("Achieved an estimated clock of " + this.preciseStr(report["clockEstim"], 2) +
+            "ns (" + this.preciseStr(report["fmax"], 2) + "MHz)");
+        println("");
+        println("Latency of " + report["latency"] + " cycles for top function " + report["topFun"]);
+        println("Estimated execution time of " + report["execTime"] + "s");
+        println();
+        println("Resource usage:");
+        println("FF:   " + report["FF"] + " (" + this.preciseStr(report["perFF"], 2) + "%)");
+        println("LUT:  " + report["LUT"] + " (" + this.preciseStr(report["perLUT"], 2) + "%)");
+        println("BRAM: " + report["BRAM"] + " (" + this.preciseStr(report["perBRAM"], 2) + "%)");
+        println("DSP:  " + report["DSP"] + " (" + this.preciseStr(report["perDSP"], 2) + "%)");
+        println("----------------------------------------");
```

Comment:
Maybe this could also benefit from a template literal

Reply:
Done